scripts/prerelease/generate_prerelease_script.py

```diff
@@ -77,6 +78,12 @@ def main(argv=sys.argv[1:]):
     build_files = get_source_build_files(config, args.rosdistro_name)
     build_file = build_files[args.source_build_name]
 
+    # Use targets defined by corresponding Release config instead of
+    # targets defined by Source config. This is to increase the number of
+    # supported targets. New targets are explicitly passed to configure_devel_job().
+    release_build_files = get_release_build_files(config, args.rosdistro_name)
+    release_build_file = release_build_files[args.source_build_name]
```

Comment:
@dirk-thomas 

`generate_prerelease_script.py` is called with the following parameterts:

```
generate_prerelease_script.py \
  <config_url:https://raw.githubusercontent.com/ros-infrastructure/ros_buildfarm_config/master/index.yaml> \
  <ros_distro:jade> <build_name:default> <os_name:ubuntu> <os_code_name:vivid> <arch:amd64> \
  <pkg_name:ardrone_autonomy> \
  --level 1 \
  --output-dir ./
```

As far as I understood, the script first parses the index file at `config_url` and looks up the `ros_distro` key. Next it looks up `ros_distro/source_builds` key and fetches the YAML config file specified by `ros_disto/source_builds/build_name`. Finally in that config file, it check if `targets/os_name/os_code_name/arch` exists. The problem I am trying to solve here is when the fetched YAML file from `ros_distro/source_builds/build_name` does not define all the targets that exist in the YAML file specified by `ros_distro/release_builds/build_name` key. 

Could you please give some examples of other failure cases? One other failure case that comes to my mind is when the user sets `build_name` to something that is only defined in the release build config file. In case of default index files, `build_name:armhf, arch:armhf` will break the pre-release script.